### ISSUE TITLE
Enhancement: Add Hessian-based Geometric Processing for SCN-PDE Solver

### ISSUE THREAD

**Issue by tsatn** [tsatn/mp-pde-scnn#5]
## Overview
Add Hessian matrix computation to enhance geometric understanding of PDE solutions using simplicial structure.

## Motivation
1. Current implementation lacks explicit geometric information about solution surface
2. Hessian matrix provides crucial second-order derivative information
3. Can improve solution accuracy and stability through geometric regularization

## Proposed Changes
1. add a class 
```
class HessianOperator(nn.Module):
    def compute_hessian(self, x, B1, B2=None):
        # compute second-order derivatives
        # use simplicial structure for geometric properties
        # return Hessian matrix
```
2. integrate into the training loop: 
- # add geometric regularization

3. update model architecture to use geometric information
